### PR TITLE
Scrapper, Anna Azimkova - 22FPL1

### PR DESCRIPTION
**Pull request by puglyann**

### CODE REVIEW

**Comment by dmitry-uraev** [comment]
No activity in PR. It is right time to start working on it.

File: lab_5_scrapper/scrapper.py [hunk 1]
```diff
@@ -187,6 +346,9 @@ def unify_date_format(self, date_str: str) -> datetime.datetime:
         Returns:
             datetime.datetime: Datetime object
         """
+        dt_object = datetime.datetime.strptime(date_str, '%d.%m.%Y')
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py:349]: just return the result directly

File: lab_5_scrapper/scrapper.py [hunk 2]
```diff
@@ -168,14 +298,43 @@ def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
...
+        if all_body:
+            all_divs = all_body[0].find_all('div', class_='entry-content entry clearfix')
+            texts = []
+            for div_bs in all_divs:
+                texts.append(div_bs.text)
+
+        self.article.text = ''.join(texts)
 
-    def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
+    def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> Article:
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py:312]: you should not modify interface - article should be available as a class field from any Parser method

File: lab_5_scrapper/scrapper.py [hunk 3]
```diff
@@ -127,11 +235,28 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        url = ''
+        for div in article_bs.find_all('div', {'class': 'post-details'}):
+            for urls in div.select('a'):
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py:240]: this method should extract just one url from one beautifulSoup element given, so why you try to iterate over a list of urls? it seems to me that you just save the last url and rewrite previous urls every iteration

File: lab_5_scrapper/scrapper.py [hunk 4]
```diff
@@ -116,6 +221,9 @@ def __init__(self, config: Config) -> None:
         Args:
             config (Config): Configuration
         """
+        self.config = config
+        self.urls = []
+        self.url_pattern = 'https://new-science.ru'
```
Comment (dmitry-uraev) [inline lab_5_scrapper/scrapper.py:226]: is it possible to extract this pattern from one of your seed urls during initialization?

**Comment by dmitry-uraev** [comment]
No significant changes since last review. And do not forget to respond to my comments

File: lab_5_scrapper/scrapper.py (hunk 1 above)
Comment (puglyann) [reply lab_5_scrapper/scrapper.py]: done

File: lab_5_scrapper/scrapper.py (hunk 2 above)
Comment (puglyann) [reply lab_5_scrapper/scrapper.py]: done :(

File: lab_5_scrapper/scrapper.py (hunk 4 above)
Comment (puglyann) [reply lab_5_scrapper/scrapper.py]: it is possible, done

**Pull request closed without merging**